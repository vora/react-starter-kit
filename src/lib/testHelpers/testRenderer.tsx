// This helper is a custom implementation of
// React Test Renderer that provides
// Components a memory router as well as
// Any additional providers that you specify.

import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import TestRenderer, {
  ReactTestRenderer,
  TestRendererOptions,
} from 'react-test-renderer';

function testRenderer(
  nextElement: ReactElement,
  options?: TestRendererOptions,
): ReactTestRenderer {
  return TestRenderer.create(
    <MemoryRouter>{nextElement}</MemoryRouter>,
    options,
  );
}

export default testRenderer;
