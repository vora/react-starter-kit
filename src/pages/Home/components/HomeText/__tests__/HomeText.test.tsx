import { HomeText } from '..';

import testRenderer from '~/lib/testHelpers/testRenderer';

describe('HomeText', () => {
  test('it renders and matches snapshot', () => {
    const wrapper = testRenderer(<HomeText />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
