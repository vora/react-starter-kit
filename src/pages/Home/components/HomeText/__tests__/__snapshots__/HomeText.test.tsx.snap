// Vitest Snapshot v1

exports[`HomeText > it renders and matches snapshot 1`] = `
<h1>
  Welcome Home
</h1>
`;
